### PR TITLE
AEA-433 Added Windows to the CI build

### PR DESCRIPTION
**Pull request by cyenyxe**
## Proposed changes

Added Windows build to CI, with the initial intention of detecting issues on this OS.

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [x] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [x] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have checked that the documentation about the `aea cli` tool works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: aea/connections/stub/connection.py [hunk 1]
```diff
@@ -20,7 +20,7 @@
 """This module contains the stub connection."""
 
 import asyncio
-import fcntl
+# import fcntl
```
Comment (DavidMinarsch) [inline aea/connections/stub/connection.py:23]: Probably incomplete fix?

File: aea/connections/stub/connection.yaml [hunk 2]
```diff
@@ -7,7 +7,7 @@ license: Apache-2.0
 aea_version: '>=0.3.0, <0.4.0'
 fingerprint:
   __init__.py: QmWwepN9Fy9gHAp39vUGFSLdnB9JZjdyE3STnbowSUhJkC
-  connection.py: QmT6Cs8oTSoZMoL1eLMYZAF1MibXz9pAKcnHTDDmtmXvxj
+  connection.py: QmXZ8QQzdJufp4uqJyQjDDkrNckmTvmuiAiAxpG8vhsomw
```
Comment (DavidMinarsch) [inline aea/connections/stub/connection.yaml:10]: We will need to bump its version and all the references to it..so all agents in packages etc

File: aea/helpers/file_lock.py [hunk 3]
```diff
@@ -0,0 +1,57 @@
...
+#   See the License for the specific language governing permissions and
+#   limitations under the License.
+#
+# ------------------------------------------------------------------------------
+
+"""Patch of 'fnctl' to make it compatible with Windows."""
+
+import os
+
+# needs win32all to work on Windows
```
Comment (DavidMinarsch) [inline aea/helpers/file_lock.py:25]: Does that need to be added as a dependency for windows only?

File: aea/skills/tasks.py [hunk 4]
```diff
@@ -116,6 +116,7 @@ def init_worker() -> None:
     :return: None
     """
     signal.signal(signal.SIGINT, signal.SIG_IGN)
+    signal.signal(signal.CTRL_C_EVENT, signal.SIG_IGN)
```
Comment (DavidMinarsch) [inline aea/skills/tasks.py:119]: why not use your new helper method?

File: scripts/generate_ipfs_hashes.py [hunk 5]
```diff
@@ -206,7 +206,7 @@ def __enter__(self):
 
     def __exit__(self, exc_type, exc_val, exc_tb):
         # terminate the ipfs daemon
-        self.process.send_signal(signal.SIGINT)
+        self.process.send_signal(signal.SIGTERM)
```
Comment (DavidMinarsch) [inline scripts/generate_ipfs_hashes.py:209]: is that cross-platform compatible then?

File: tests/test_cli/test_add_key.py [hunk 6]
```diff
@@ -176,7 +177,8 @@ def setup_class(cls):
             cls.agent_folder / ETHEREUM_PRIVATE_KEY_FILE,
         )
 
-    def test_add_many_keys(self):
+    @skip_test_windows(is_class_test=True)
```
Comment (DavidMinarsch) [inline tests/test_cli/test_add_key.py:180]: Why do they fail on windows?

File: tests/test_cli/test_gui.py [hunk 7]
```diff
@@ -61,6 +63,7 @@ def setup_class(cls):
         )
         time.sleep(10.0)
 
+    @skip_test_windows(is_class_test=True)
```
Comment (DavidMinarsch) [inline tests/test_cli/test_gui.py:66]: Why is it failing?

File: aea/skills/tasks.py (hunk 4 above)
Comment (marcofavorito) [reply aea/skills/tasks.py:119]: it is not "send_signal". This just ignores the `CTRL_C_EVENT` signal on Windows.

File: scripts/generate_ipfs_hashes.py (hunk 5 above)
Comment (marcofavorito) [reply scripts/generate_ipfs_hashes.py:209]: yes. For some reason, `process.send_signal` does not accept `SIGINT`:

https://github.com/python/cpython/blob/3.8/Lib/subprocess.py#L1422-L1432

File: packages/fetchai/connections/p2p_noise/connection.py [hunk 8]
```diff
@@ -331,8 +331,9 @@ async def start(self) -> None:
             os.remove(in_path)
         if os.path.exists(out_path):
             os.remove(out_path)
-        os.mkfifo(in_path)
-        os.mkfifo(out_path)
+        # Ignore type-hinting check for Windows
+        os.mkfifo(in_path)  # type: ignore
+        os.mkfifo(out_path)  # type: ignore
```
Comment (marcofavorito) [inline packages/fetchai/connections/p2p_noise/connection.py:336]: this needs to be fixed at some point. `os.mkfifo` is not available on Windows

File: aea/skills/tasks.py [hunk 9]
```diff
@@ -116,6 +116,7 @@ def init_worker() -> None:
     :return: None
     """
     signal.signal(signal.SIGINT, signal.SIG_IGN)
+    # signal.signal(signal.CTRL_C_EVENT, signal.SIG_IGN)
```
Comment (DavidMinarsch) [inline aea/skills/tasks.py:119]: ?

File: aea/helpers/exec_timeout.py [hunk 10]
```diff
@@ -150,6 +152,8 @@ def _set_timeout_watch(self) -> None:
 
         :return: None
         """
+        if os.name == "nt":
+            pytest.skip("signal.settimer non available on Windows.")
```
Comment (DavidMinarsch) [inline aea/helpers/exec_timeout.py:156]: Hmm surely we don't just want to skip during testing?

File: packages/fetchai/connections/p2p_noise/connection.py (hunk 8 above)
Comment (DavidMinarsch) [reply packages/fetchai/connections/p2p_noise/connection.py:336]: #1204

File: aea/helpers/exec_timeout.py (hunk 10 above)
Comment (marcofavorito) [reply aea/helpers/exec_timeout.py:156]: yes my bad, will fix now

File: aea/skills/tasks.py (hunk 9 above)
Comment (marcofavorito) [reply aea/skills/tasks.py:119]: uncommented

**Pull request merged**